pkg/core/combat/enemy.go

```diff
@@ -107,12 +96,15 @@ func (c *Handler) EnemyByDistance(x, y float64, excl TargetKey) []int {
 }
 
 // EnemiesWithinRadius returns an array of indices of the enemies within radius r
-func (c *Handler) EnemiesWithinRadius(x, y, r float64) []int {
+func (c *Handler) EnemiesWithinRadius(p Point, r float64) []int {
 	result := make([]int, 0, len(c.enemies))
 	for i, v := range c.enemies {
-		vx, vy := v.Shape().Pos()
-		dist := math.Pow(x-vx, 2) + math.Pow(y-vy, 2)
-		if dist > r {
+		vPos := v.Shape().Pos()
+		dist := p.Sub(vPos).MagnitudeSquared()
+		if dist > r*r {
+			continue
+		}
```

Comment:
Another place where we should make sure we align on the `==` case for collision checks

Also, wonder if we should be doing a full on circle-circle collision check with all enemies here? With the current logic, we will only return enemies with their **center** inside of the given radius

Reply:
>Also, wonder if we should be doing a full on circle-circle collision check with all enemies here 
Yes, we should. That will part of another PR.

Reply:
No further change required, because `==` means `colliding`/`true`.